Sources/Common/Util/RingBuffer.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/// A ring buffer (or circular buffer) is a fixed-size, array-based data structure
+/// that is efficient for FIFO (First-In-First-Out) operations.
+struct RingBuffer<T> {
```

Comment:
I tried looking into any default data structure that might exist for limited-sized FIFO and could't find any. 

I did think about using limited sized array, instead of creating a double link list but then decided RingBuffer because it provides `O(1)` efficiency for insertions and deletions, unlike a standard array that requires `O(n)` operations due to element shifting.

It seamlessly handles fixed capacity with automatic data overwriting and simplifies code by encapsulating FIFO logic. It's initialized with a fixed size so it allocates exactly as much memory as required but shrinking and expanding could lead to memory wastage. 

So even if they seem like small wins, i felt like it would be nice to have a cleaner solution and honestly implementation was very simple so had alot of confidence and ofcource test suite is also present 😄